openedx/openedx-platform
# feat(mobile_api): Add course access object to mobile course info API

## Description
This PR is a follow-up to FC-0031 project for Mobile API updates https://github.com/openedx/edx-platform/issues/33304, and extends `BlocksInfoInCourseView` view with `courseware_access_details` data for a user in the form:
```
"course_access_details": {
		"has_unmet_prerequisites": <bool>,
		"is_too_early": <bool>,
		"is_staff": <bool>,
		"audit_access_expires": null,
		"courseware_access": {
			"has_access": <bool>,
			"error_code": <string> or null,
			"developer_message":  <string> or null,
			"user_message":  <string> or null,
			"additional_context_user_message":  <string> or null,
			"user_fragment": null
		}
	},
	
"enrollment_details": {
		"created": <date>,
		"mode": <string>,
		"is_active": <bool>
	},

"course_updates": <str>,
"course_handouts": <str>,
"discussion_url": <str> or null,
	
"course_sharing_utm_parameters": {
		"facebook": "utm_medium=social&utm_campaign=social-sharing-db&utm_source=facebook",
		"twitter": "utm_medium=social&utm_campaign=social-sharing-db&utm_source=twitter"
	},
"course_modes": [
		{
			"slug": <str>,
			"sku": null,
			"android_sku": null,
			"ios_sku": null,
			"min_price": <int>
		}
	]
```
The `coursewareAccess` data should be provided only if it is requested by a staff or admin user, or by the users themselves.

## Supporting information
This PR further extends the mobile `BlocksInfoInCourseView` that was introduced in the https://github.com/openedx/edx-platform/pull/33296

## Testing instructions
Run a **GET** `/api/mobile/{api_version}/course_info/blocks/?course_id=<course_id>&username=<username>&return_type=dict`

Asses the response, check that fields "course_access_details" and "certificate" are in the response body.

## Deadline

End of the month, as those changes are required to unblock the further works around mobile courseware enhancements.
